system/lib/compiler-rt/lib/asan/asan_flags.cc

```diff
@@ -120,6 +130,26 @@ void InitializeFlags() {
   lsan_parser.ParseString(lsan_default_options);
 #endif
 
+#if SANITIZER_EMSCRIPTEN
+  char *options;
+  // Override from Emscripten Module.
+#define MAKE_OPTION_LOAD(parser, name) \
+    options = (char*) EM_ASM_INT({ \
+      return _emscripten_with_builtin_malloc(function () { \
+        return allocateUTF8(Module[name] || 0); \
+      }); \
+    }); \
+    parser.ParseString(options); \
+    emscripten_builtin_free(options);
```

Comment:
Could this be a stack allocation instead of a heap allocation?

Reply:
I suppose we would need a new version of `allocateUTF8` that allocates on the stack. If that doesn't exist it might be good to add.

Reply:
```
function preserveStack(func) {
  try {
    var stack = stackSave();
    return func();
  } finally {
    stackRestore(stack);
  }
}
```

Reply:
Or not, since I guess the JS needs to be able to return the pointer :+1: